src/jormungandr/jormungandr/src/network/subscription.rs

```diff
@@ -288,6 +245,338 @@ impl Sink<net_data::Header> for BlockAnnouncementProcessor {
+                            )
+                        }
+                        Err(e) => {
+                            tracing::error!(
+                                node_id = %self.node_id,
+                                address = %ingress_addr,
+                                error = ?e,
+                                "failed to execute reverse DNS lookup for address"
+                            );
+
+                            (
+                                FragmentProcessorSendFragmentsState::SendFragments {
+                                    ingress_addr,
+                                    resolved_hostnames: Vec::new(),
+                                },
+                                Poll::Pending,
+                            )
+                        }
+                    }
+                }
+                Poll::Pending => (
+                    FragmentProcessorSendFragmentsState::ReverseLookup { ingress_addr, fut },
+                    Poll::Pending,
+                ),
+            },
+            FragmentProcessorSendFragmentsState::SendFragments {
+                ingress_addr,
+                resolved_hostnames,
+            } => {
+                let should_send = match self.global_state.config.whitelist.as_ref() {
+                    Some(whitelist) => {
+                        tracing::debug!(
+                            "whitelist configured, checking whether fragments should be sent"
+                        );
+
+                        whitelist.iter().any(|ma| {
+                            match ma.iter().next() {
+                                Some(multiaddr::Protocol::Ip4(addr)) => addr == ingress_addr.ip(),
+                                Some(multiaddr::Protocol::Ip6(addr)) => addr == ingress_addr.ip(),
+                                Some(multiaddr::Protocol::Dns4(fqdn))
+                                | Some(multiaddr::Protocol::Dns6(fqdn)) => {
+                                    tracing::debug!(resolved_hostnames = ?resolved_hostnames, fqdn = %fqdn, "checking whitelist for domain");
+
+                                    resolved_hostnames.iter().any(|x| x == &fqdn)
+                                }
+                                _ => {
+                                    // In this case the whitelisted multiaddr is invalid
+                                    // so consider it as not matching anything
+                                    tracing::error!(multiaddr = %ma, "Invalid entry in whitelist");
+                                    false
+                                }
+                            }
+                        })
+                    }
+                    None => {
+                        tracing::debug!("no whitelist configured, bypassing check");
+
+                        true
+                    }
+                };
+
+                if should_send {
+                    let fragments = self.clear_buffered_fragments();
```

Comment:
why do we flush the buffer here?

Reply:
I guess `clear_buffered_fragments` is kind of a bad name. This is what was already done in this part. It just empties the fragments buffer and sends all of them through the message box.

https://github.com/input-output-hk/catalyst-core/pull/241/files/d284ff271c9c130b5833e64a025b4528125566eb#diff-c9c45cadddc07944c423cf18d0c0775c7fda7d1c69e0ec690e21fa8bcda5e0b8L375-L378